tfe/resource_tfe_notification_configuration.go

```diff
@@ -75,9 +96,18 @@ func resourceTFENotificationConfiguration() *schema.Resource {
 			},
 
 			"workspace_external_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
```

Comment:
This is all looking awesome! There's only one thing that I can't quite map out in my head, and we'll use Notification Configurations here as an example.

Notification Configurations require a workspace  (as the `Required: true` here enforces), but in this migration you're (having to) mark both of these as both optional and `Computed`, to allow the fields to be saved to state. 

* What does that look like then, when you try and create a fresh new Notification Configuration without the workspace? In the new case, not one that's existing and migrating.
* What happens later after all this migration stuff is completed? Will `workspace_id` revert to `Required: true, ForceNew: true, Computed: false`? Is that problematic in any way...?

Reply:
> What does that look like then, when you try and create a fresh new Notification Configuration without the workspace? In the new case, not one that's existing and migrating.

if you do this, you'll get an error on apply. normally you'd get the error on plan but since i need to allow both attributes temporarily, i have to validate them myself later.
although i've set it to optional and computed, i still wrote checks on create that require workspace_id or workspace_external_id to be set. 
the error looks like this:
```
Error: One of workspace_id or workspace_external_id must be set

  on main.tf line 51, in resource "tfe_notification_configuration" "notification_config":
  51: resource "tfe_notification_configuration" "notification_config" {
```

Reply:
Perfect!

Reply:
> What happens later after all this migration stuff is completed? Will workspace_id revert to Required: true, ForceNew: true, Computed: false? Is that problematic in any way...?

yep! workspace_id will revert to Required: true, ForceNew: true, and Computed: false. i don't think this should cause problems because:
1. Required: although the attribute isn't required at the schema level, it is still required to be set in the schema for creation. 
1. ForceNew: the custom diff checks ensure that ForceNew is true right now in all cases except when the user is migrating from workspace_external_id to workspace_id (or back the other direction)
1. Computed: this one is harder for me to articulate clearly, but right now, these are only computed so that both attributes get populated on read, that way anyone migrating won't get a plan/apply that tells them they have changes when they really don't.